src/shm_ext.c

```diff
@@ -1786,10 +1814,11 @@ sr_shmext_change_sub_check(sr_conn_ctx_t *conn, sr_mod_t *shm_mod, sr_datastore_
                 }
                 has_locks = SR_LOCK_WRITE;
 
-                /* EXT WRITE LOCK */
-                if ((err_info = sr_shmext_conn_remap_lock(conn, SR_LOCK_WRITE, 0, __func__))) {
+                /* EXT READ LOCK */
+                if ((err_info = sr_shmext_conn_remap_lock(conn, SR_LOCK_READ, 1, __func__))) {
                     goto changesub_unlock;
                 }
+                ext_lock = 1;
```

Comment:
I think this is not correct (but was not even before). On the `goto`, you will be unlocking REMAP lock even though it is not locked. I suppose if you change line 1855 to `if (has_locks && ext_lock) {`, it would work correctly. Similarly for the following changes.

Reply:
Right, I completely missed that there are 2 labels... Okay then.